### PR TITLE
ENT-3963 Fix self upgrade on AIX

### PR DESCRIPTION
**Pull request by nickanderson**

### CODE REVIEW

File: standalone_self_upgrade.cf.in [hunk 1]
```diff
@@ -948,6 +948,9 @@ body package_method u_generic(repo)
 
       package_delete_command     => "/usr/sbin/installp -ug cfengine.cfengine-nova$";
 
+      # Internal version comparison model doesn't work for W.X.Y.Z
+        package_version_less_command => "$(sys.bindir)/rpmvercmp '$(v1)' lt '$(v2)'";
+        package_version_equal_command => "$(sys.bindir)/rpmvercmp '$(v1)' eq '$(v2)'";
```
Comment (vpodzime) [inline standalone_self_upgrade.cf.in]: Weird indentation here.

File: standalone_self_upgrade.cf.in [hunk 2]
```diff
@@ -155,8 +155,10 @@ bundle agent cfengine_software
       "$(sys.statedir)/MPF/."
         create => "true";
 
+      "$(sys.statedir)/MPF/desired-cfengine-package-version.json"
+        create => "true";
+
       "$(sys.statedir)/MPF/desired-cfengine-package-version.json" -> { "ENT-3592", "ENT-3937" }
-        create => "true",
         edit_line => cfengine_software_version_data;
```
Comment (vpodzime) [inline standalone_self_upgrade.cf.in]: Isn't this a bug? I'd expect this change to make no difference.

Comment (nickanderson) [reply standalone_self_upgrade.cf.in]: I must have patched this upstream already. It's true that there should be no functional difference when separated into its own promise or not. Best practice does dictate to separate into smallest parts so I am inclined to leave this further separation in place.

File: standalone_self_upgrade.cf.in (hunk 1 above)
Comment (nickanderson) [reply standalone_self_upgrade.cf.in]: re-indented

**Pull request merged**